crates/core/src/transport/connection_handler.rs

```diff
@@ -125,11 +125,11 @@ impl OutboundConnectionHandler {
         bandwidth_limit: Option<usize>,
     ) -> Result<(Self, mpsc::Receiver<PeerConnection>), TransportError> {
         // Channel buffer is one so senders will await until the receiver is ready, important for bandwidth limiting
-        let (conn_handler_sender, conn_handler_receiver) = mpsc::channel(1000);
-        let (new_connection_sender, new_connection_notifier) = mpsc::channel(1000);
+        let (conn_handler_sender, conn_handler_receiver) = mpsc::channel(100);
```

Comment:
TODO: I would experiment to set this to 1 and see if we still have problems, because we shouldn't (we can revert back to 100, but ensuring that it works with a capacity of just 1 would be good, or if not try find out experimentally at what point it breaks down and dig into why)

Reply:
Excellent suggestion\! Setting the channel capacity to 1 would be a great stress test to ensure our connection handling is robust. I'll experiment with this - it should help us identify any race conditions or improper assumptions about buffering. If it breaks at capacity 1, we can incrementally increase to find the minimum viable buffer size and understand why that specific threshold is needed.